pkg/clients/service.go

```diff
@@ -0,0 +1,72 @@
+package clients
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/monitoring"
+	hClient "github.com/ory/hydra-client-go/v2"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	hydra HydraClientInterface
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+func (s *Service) parseServiceError(r *http.Response, e error) *hClient.ErrorOAuth2 {
```

Comment:
took me a while to understand that the hydra client doesn't return error objects, mind adding a small description or changing the name of the method?